README.md

```diff
@@ -13,312 +13,252 @@ to [dev.poktroll.com/category/actors](https://dev.poktroll.com/category/actors).
-  - [Error Handling](#error-handling)
-  - [Dependencies implementation](#dependencies-implementation)
-  - [Poktroll dependencies](#poktroll-dependencies)
+  - [Interface Design](#interface-design)
+  - [ShannonSDK components](#shannonsdk-components)
+    - [Account Client](#account-client)
+    - [Application Client](#application-client)
+    - [Application Ring](#application-ring)
+    - [Block Client](#block-client)
+    - [Signer](#signer)
+    - [Session Client](#session-client)
+    - [Filtered Session](#filtered-session)
+    - [Relayer](#relayer)
 
 ## Overview
 
-ShannonSDK encapsulates various clients and a signer necessary for interacting
+ShannonSDK encapsulates various structures and a functions necessary for interacting
 with the `Poktroll` network. It provides an intuitive interface to manage `Sessions`,
-and `RelayRequests`.
-
-This document outlines the key components and functionalities of the SDK, along
-with detailed usage instructions.
+`Applications`, and `RelayRequests`.
 
 ## Key Components
 
 The SDK consists of the following core components:
 
-- **ApplicationClient**: Handles interactions related to applications on the network.
-- **SessionClient**: Manages session-related operations.
-- **AccountClient**: Deals with account-related queries and operations.
-- **SharedParamsClient**: Provides shared parameters such as various governance params to the SDK.
+- **Relay handling**: Building and validating RelayRequests and RelayResponses.
+- **AccountClient**: Handles account-related queries and operations.
+- **ApplicationClient**: Manages application-related operations and queries.
+- **ApplicationRing**: Manages the list of gateways delegations from applications
+and handling of ring signatures.
 - **BlockClient**: Fetches information about blocks on the network.
-- **RelayClient**: Sends relay requests to the network.
+- **SessionClient**: Manages session-related operations.
 - **Signer**: Signs relay requests to ensure authenticity and integrity.
 
 ## Initialization
 
-To create a new instance of ShannonSDK, you need to provide the implementations for
-the required clients and signer. Here is an example of how to initialize the SDK:
+## Usage
+
+For a given request, the main workflow for using the ShannonSDK is the following:
+1. Get the current block height.
+2. Get the session corresponding to the block height.
+3. Select a supplier endpoint from those available in the session.
+4. Build a relay request.
+5. Sign the relay request.
+6. Send the relay request to the selected endpoint.
+7. Validate the received relay response.
+
+### Get Session
 
 ```go
-applicationClient := NewApplicationClient(grpcConn)
-sessionClient := NewSessionClient(grpcConn)
-accountClient := NewAccountClient(grpcConn)
-sharedParamsClient := NewSharedParamsClient(grpcConn)
-blockClient := NewBlockClient(poktrollRPCURL)
-relayClient := NewRelayClient()
-signer := NewSigner(privateKeyHex)
-
-sdk, err := NewShannonSDK(
```

Comment:
Just curious, was depinject considered for use here (i.e. receiving a `depinject.Config`)?

Reply:
Mainly to reduce third party dependencies and cognitive load.